tests/trx_generator/trx_provider.cpp

```diff
@@ -73,15 +73,29 @@ namespace eosio::testing {
    }
 
    void p2p_connection::disconnect() {
-      ilog("Closing socket.");
-      _p2p_socket.close();
-      ilog("Socket closed.");
+      int max    = 30;
+      int waited = 0;
+      while (_sent.load() != _sent_callback_num.load() && waited < max) {
+         ilog("disconnect waiting on ack - sent ${s} | acked ${a} | waited ${w}",
+              ("s", _sent.load())("a", _sent_callback_num.load())("w", waited));
```

Comment:
The ilog may not be accurate as we reload the `atomic`.  Maybe do something like this:
```suggestion
      for (int sent = _sent.load(), sent_callback_num = _sent_callback_num.load(); sent != sent_callback_num && waited < max; ) {
         ilog("disconnect waiting on ack - sent ${s} | acked ${a} | waited ${w}",
              ("s", sent)("a", sent_callback_num)("w", waited));
```